src/kotlin_script.cpp

```diff
@@ -206,72 +195,64 @@ PropertyInfo KotlinScript::get_class_category() const {
     // TODO/4.0:
     return {};
 }
-#endif
-
-KotlinScript::KotlinScript() : kotlin_class(nullptr) {}
 
 PlaceHolderScriptInstance* KotlinScript::placeholder_instance_create(Object* p_this) {
-#ifdef TOOLS_ENABLED
     PlaceHolderScriptInstance* placeholder {
       memnew(PlaceHolderScriptInstance(KotlinLanguage::get_instance(), Ref<Script>(this), p_this))};
-    p_this->set_script_instance(placeholder);
+
+    List<PropertyInfo> exported_properties;
+    get_script_exported_property_list(&exported_properties);
+    placeholder->update(exported_properties, exported_members_default_value_cache);
+
     placeholders.insert(placeholder);
-    _update_exports(placeholder);
     return placeholder;
-#else
-    return nullptr;
-#endif
 }
 
 void KotlinScript::update_exports() {
-#ifdef TOOLS_ENABLED
-    for (PlaceHolderScriptInstance* script_instance : placeholders) {
-        _update_exports(script_instance);
+    //TODO: Remove this when multithreading is fixed.
+    if(!Thread::is_main_thread()){
+        MessageQueue::get_singleton()->push_callable(callable_mp(this, &KotlinScript::update_exports));
+        return;
     }
-#endif
-}
 
-void KotlinScript::_update_exports(PlaceHolderScriptInstance* placeholder) {
-#ifdef TOOLS_ENABLED
     exported_members_default_value_cache.clear();
+    if (!is_valid()) { return; }
 
     Callable::CallError call;
     Object* tmp_object {_new({}, 0, call)};
     KotlinInstance* script_instance {reinterpret_cast<KotlinInstance*>(tmp_object->get_script_instance())};
 
-    List<PropertyInfo> all_properties;
-    get_script_property_list(&all_properties);
-
     List<PropertyInfo> exported_properties;
-    for (const PropertyInfo& property_info : all_properties) {
-        if (property_info.usage & PropertyUsageFlags::PROPERTY_USAGE_EDITOR) {
-            exported_properties.push_back(property_info);
-        }
-    }
+    get_script_exported_property_list(&exported_properties);
 
     for (int i = 0; i < exported_properties.size(); ++i) {
         Variant default_value;
         const String& property_name {exported_properties[i].name};
         script_instance->get_or_default(property_name, default_value);
         exported_members_default_value_cache[property_name] = default_value;
     }
-    placeholder->update(exported_properties, exported_members_default_value_cache);
+
+    for (PlaceHolderScriptInstance* placeholder : placeholders) {
+        placeholder->update(exported_properties, exported_members_default_value_cache);
+    }
+
     memdelete(tmp_object);
-#endif
 }
 
 void KotlinScript::_placeholder_erased(PlaceHolderScriptInstance* p_placeholder) {
-#ifdef TOOLS_ENABLED
     placeholders.erase(p_placeholder);
-#endif
 }
+#endif
 
 void KotlinScript::_bind_methods() {
     ClassDB::bind_vararg_method(METHOD_FLAGS_DEFAULT, "new", &KotlinScript::_new, MethodInfo("new"));
```

Comment:
Just as a question: we have this limitation: 
![2023-11-01-093650_706x250_scrot](https://github.com/utopia-rise/godot-kotlin-jvm/assets/22662033/b1a1f1d6-cb9d-4494-a8b6-47538352bba6)
Could this somehow be circumvented by registering other "new" functions with the parameters of the constructor here somehow?

Reply:
I don't know how this behaves. Should be investigated in another PR.